File: backend/timebank/src/main/java/kookmin/software/capstone2023/timebank/domain/repository/PasswordAuthenticationJpaRepository.java
package kookmin.software.capstone2023.timebank.domain.repository;

import kookmin.software.capstone2023.timebank.domain.model.auth.PasswordAuthentication;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface PasswordAuthenticationJpaRepository extends JpaRepository<PasswordAuthentication, Long> {

    Optional<PasswordAuthentication> findByUserId(Long userId);

    PasswordAuthentication findByUsername(String username);

    boolean existsByUsername(String username);
}

File: src/main/java/kookmin/software/capstone2023/timebank/domain/repository/PasswordAuthenticationJpaRepository.kt
package kookmin.software.capstone2023.timebank.domain.repository

import kookmin.software.capstone2023.timebank.domain.model.auth.PasswordAuthentication
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface PasswordAuthenticationJpaRepository : JpaRepository<PasswordAuthentication, Long> {
    fun findByUserId(userId: Long): PasswordAuthentication?
    fun findByUsername(username: String): PasswordAuthentication?
    fun existsByUsername(username: String): Boolean
}
